### PR TITLE
change: creating core resources does not allow passing in `create_time` and `update_time`

### LINKED ISSUE #10094: remove create_time and update_time for resources in admin api
# Summary information

1. Documentation: Remove the field description that needs to pass in the creation time and update time when creating.
2. delete update_time and create_time from define of admin API .
3. Certificate information is parsed from the certificate body without additional storage.

Actually. we need confrim all of the core resource.

- [x] consumer
- [x] consumer_group
- [x] discovery_nodes
- [x] global_rule
- [x] plugin_config
- [x] plugins
- [x] proto
- [x] route
- [x] service
- [x] ssl
  validity_end
  validity_start
  exptime
- [x] stream_route
- [x] upstream

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

Fixes #10094

email: https://lists.apache.org/thread/968kp7hd6zcg7ty2clomkbshmd53v71d

this PR depend on https://github.com/apache/apisix/issues/10229 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

@monkeyDluffy6017 @leslie-tsang pls help me to review

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
This pr needs to fix the issue first: https://github.com/apache/apisix/issues/10229

**Review by yuluo-yx — APPROVED** [review]
LGTM.

File: apisix/admin/resource.lua [hunk 1]
```diff
@@ -49,7 +49,41 @@ local function split_typ_and_id(id, sub_path)
 end
 
 
-function _M:check_conf(id, conf, need_id, typ)
+local function check_create_update_time(conf)
```
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:52]: Why do you need to differentiate each resource?
Why not do like the function `utils.iinject_timestamp`?

Comment (Sn0rt) [reply apisix/admin/resource.lua:52]: Logically speaking, it is not allowed to pass in create_update and update_time when creating any core resources. I just expanded the check scope of the check_conf function to comply with the original intention of this function design (I understand)

Reason for not modifying `inject_timestamp`
1. This function does not check the subsource design. For example, when creating a route resource, the route resource contains the upstream resource. At this time, the upstream resource has create_time.

File: apisix/admin/resource.lua [hunk 2]
```diff
@@ -49,7 +49,41 @@ local function split_typ_and_id(id, sub_path)
...
+    local not_allow_create_time = "forbidden create_time in request body"
+    local not_allow_update_time = "forbidden update_time in request body"
```
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:54]:
```suggestion
    local not_allow_create_time = "the property is forbidden: create_time"
    local not_allow_update_time = "the property is forbidden: update_time"
```

File: apisix/admin/resource.lua (hunk 1 above)
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:52]:
```suggestion
local function check_forbidden_properties(conf)
```

File: apisix/admin/resource.lua [hunk 3]
```diff
@@ -49,7 +49,41 @@ local function split_typ_and_id(id, sub_path)
...
+            return not_allow_update_time
+        end
+    end
+
+    if conf.plugins then
+        if conf.plugins.create_time then
+            return not_allow_create_time
+        end
+        if conf.plugins.update_time then
+            return not_allow_update_time
```
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:78]: Is it possible to use recursion to check all properties?

File: apisix/admin/resource.lua [hunk 4]
```diff
@@ -49,7 +49,41 @@ local function split_typ_and_id(id, sub_path)
...
 
 
-function _M:check_conf(id, conf, need_id, typ)
+local function check_create_update_time(conf)
+    local not_allow_create_time = "forbidden create_time in request body"
+    local not_allow_update_time = "forbidden update_time in request body"
+
+    if conf.create_time then
+        return not_allow_create_time
+    end
```
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:58]: Is it possible to define a check list and sequentially check for properties that should not exist? like
```
local forbidden_properties = {create_time, update_time}
```

File: apisix/admin/resource.lua [hunk 5]
```diff
@@ -49,7 +49,41 @@ local function split_typ_and_id(id, sub_path)
...
+        if conf.plugins.update_time then
+            return not_allow_update_time
+        end
+    end
+
+    return nil
+end
+
+
+function _M:check_conf(id, conf, need_id, need_time, typ)
```
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:86]: Is it better to put `need_time` in the last position?

File: apisix/admin/resource.lua [hunk 6]
```diff
@@ -139,7 +181,7 @@ function _M:post(id, conf, sub_path, args)
         return 405, {error_msg = "not supported `POST` method for " .. self.kind}
     end
 
-    local id, err = self:check_conf(id, conf, false)
+    local id, err = self:check_conf(id, conf, false, false)
```
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:184]: You don't need to pass false, nil is ok

File: apisix/admin/resource.lua (hunk 5 above)
Comment (monkeyDluffy6017) [inline apisix/admin/resource.lua:86]: Is `forbidden_time` better ?

File: apisix/admin/resource.lua (hunk 2 above)
Comment (Sn0rt) [reply apisix/admin/resource.lua:54]: looks good , I will fix it

File: apisix/admin/resource.lua (hunk 5 above)
Comment (Sn0rt) [reply apisix/admin/resource.lua:86]: ok. I will change it to `allow_time`

File: apisix/admin/resource.lua (hunk 6 above)
Comment (Sn0rt) [reply apisix/admin/resource.lua:184]: sound good. I will fix it later

File: apisix/admin/resource.lua (hunk 5 above)
Comment (Sn0rt) [reply apisix/admin/resource.lua:86]: great!

File: apisix/admin/resource.lua (hunk 4 above)
Comment (Sn0rt) [reply apisix/admin/resource.lua:58]: great! I will fix it later

File: apisix/admin/resource.lua (hunk 3 above)
Comment (Sn0rt) [reply apisix/admin/resource.lua:78]: I referred to other definitions containing subresources, such as ssl, and I found the benefits of this implementation

1. Simple to implement, easy to maintain, no recursion required.
2. Currently, only these two sub-resources are included in other resources.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**